docs/concepts/function-components/attribute.md

```diff
@@ -41,7 +41,9 @@ pub fn rendered_at(props: &RenderedAtProps) -> Html {
 #[function_component(App)]
 fn app() -> Html {
     let (counter, set_counter) = use_state(|| 0);
-    let onclick = Callback::from(move |_| set_counter(*counter + 1));
+
+    let counter1 = counter.clone();
+    let onclick = Callback::from(move |_| set_counter(*counter1 + 1));
```

Comment:
Ah right, should've realised that. Thanks for checking!

One little trick I like to use that makes this a bit prettier is this:
```rust
let onclick = {
  // also note the use of `Rc::clone` which is how Rust recommends you clone `Rc`s
  let counter = Rc::clone(counter);
  Callback::from(move |_| set_counter(*counter + 1))
};
```

It's slightly more verbose and might require some getting used to but this has a few benefits:
- no need to come up with a new variable name every time
- It's immediately clear that the cloning and callback are related
- the cloned variable's scope is limited to the scope so you can't accidentally use it somewhere else

Reply:
Huh, is there a reason why `Rc::clone(&counter)` is better than `counter.clone()`?

Reply:
Because `.clone` doesn't tell you much about what's actually happening. Cloning an `Rc` is comparably cheap and is linear in complexity, something that isn't true for strings or vectors. It usually makes sense to optimize away cloning as much as possible but that isn't necessary for `Rc`s and by using `Rc::clone(_)` we make it explicit that that's what we're doing.

And also, from the official docs:
> The Rc::clone(&from) syntax is the most idiomatic because it conveys more explicitly the meaning of the code. In the example above, this syntax makes it easier to see that this code is creating a new reference rather than copying the whole content of foo.

Reply:
That's good to know